internal/commands/app/create_test.go

```diff
@@ -335,75 +330,59 @@ Check out your app: cd ./test-app && realm-cli app describe
-			return []realm.Template{
-				{
-					ID:   "palm-pilot.bitcoin-miner",
-					Name: "Mine bitcoin on your Palm Pilot from the comfort of your home, electricity not included",
-				},
-				{
-					ID:   "blackberry.important-business-app",
-					Name: "Oh wow, a Blackberry... you must a very powerful, extravagant man.",
-				},
-			}, nil
+		testApp := realm.App{
+			ID:          "456",
+			GroupID:     "123",
+			ClientAppID: "bitcoin-miner-abcde",
+			Name:        "bitcoin-miner",
 		}
 
-		clientZipPkg, err := zip.OpenReader("testdata/react-native.zip")
+		backendZipPkg, err := zip.OpenReader("testdata/project.zip")
 		assert.Nil(t, err)
-		rc.ClientTemplateFn = func(groupID, appID, templateID string) (*zip.Reader, bool, error) {
-			return &clientZipPkg.Reader, true, nil
-		}
-		ac := mock.AtlasClient{}
-		ac.GroupsFn = func() ([]atlas.Group, error) {
-			return []atlas.Group{{ID: "123"}}, nil
-		}
-		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
-			return []atlas.Cluster{{Name: "test-cluster"}}, nil
+
+		client := mock.RealmClient{
+			FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+				return []realm.App{}, nil
+			},
+			ExportFn: func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
+				return "", &backendZipPkg.Reader, nil
+			},
+			CreateAppFn: func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+				return realm.App{
+					GroupID:     groupID,
+					ID:          "456",
+					ClientAppID: name + "-abcde",
+					Name:        name,
+					AppMeta:     meta,
+				}, nil
+			},
+			ImportFn: func(groupID, appID string, appData interface{}) error {
+				return nil
+			},
 		}
 
 		cmd := &CommandCreate{createInputs{
 			newAppInputs: newAppInputs{
-				Name:            "template-app",
-				Location:        realm.LocationVirginia,
+				Name:            "bitcoin-miner",
+				Project:         testApp.GroupID,
+				Location:        realm.LocationIreland,
 				DeploymentModel: realm.DeploymentModelGlobal,
-				ConfigVersion:   realm.DefaultAppConfigVersion,
 			},
 			Clusters: []string{"test-cluster"},
 		}}
 
-		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: ac}))
-
-		console.Tty().Close() // flush the writers
-		<-doneCh              // wait for procedure to complete
+		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: client, Atlas: mock.AtlasClient{
+			ClustersFn: func(groupID string) ([]atlas.Cluster, error) {
+				return []atlas.Cluster{{Name: "test-cluster"}}, nil
+			},
+		}}))
 
-		path := filepath.Join(profile.WorkingDirectory, cmd.inputs.Name, local.BackendPath)
-		appLocal, err := local.LoadApp(path)
+		_, err = local.LoadApp(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
 		assert.Nil(t, err)
 
-		assert.Equal(t, appLocal.RootDir, path)
+		backendFileInfo, err := ioutil.ReadDir(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
+		assert.Nil(t, err)
+		assert.Equal(t, len(backendFileInfo), 10)
```

Comment:
was this an existing assert?  do you mind at least providing some clarity into where the `10` comes from?  (and/or is there a better/more specific assert we can use instead of this?)

Reply:
ah yeah, that was an existing change that I updated. This 10 comes from the mock backend zip package which is the contents of `testdata/project.zip`.

This test serves to ensure that when a user does not supply --template, a default app will be created.

the assertion that the resulting directory has len = 10 is just to ensure that the result is the same as the result of the mocked export function.

maybe we can unzip this `testdata/project.zip` first and use this as the assertion value

ie:
```go
backendZipPkg, err := zip.OpenReader("testdata/project.zip")
assert.Nil(t, err)

// add this
expectedBackendZipLen := len(backendZipPkg.File)

realmClient := mock.RealmClient{
    ExportFn: func(groupID, appID string, req realm.ExportRequest) (string, *zip.Reader, error) {
        return "", &backendZipPkg.Reader, nil
    },
    // ...
}

// ...

// and use the new expected length here
assert.Equal(t, len(backendFileInfo), expectedBackendZipLen)
```

Reply:
edit nevermind, we cant do this. the zip file also lists out nested files so the number of files in the zip file != the number of `backendFileInfo` files.

I think it's probably sufficient to assert that:

1. we do not create a `backend` and `frontend` directory in the top level
2. and that realm_config.json exists in the top level to denote that this directory is a realm app folder

I'll do this for now, but I'm happy to change this if you have any opinions here